helix-view/src/document.rs

```diff
@@ -29,9 +29,9 @@ pub const SCRATCH_BUFFER_NAME: &str = "[scratch]";
 
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
 pub enum Mode {
-    Normal,
-    Select,
-    Insert,
+    Normal = 0,
+    Select = 1,
+    Insert = 2,
 }
```

Comment:
Do we really need this?

Reply:
Yes we do, because we use it to do array indexing.

Reply:
Aren't these the default discriminant values?

https://doc.rust-lang.org/reference/items/enumerations.html#custom-discriminant-values-for-fieldless-enumerations

Reply:
Yeah they're the default but I put them anyway to avoid confusion.